flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/utils/PythonUtil.scala

```diff
@@ -40,6 +42,15 @@ object PythonUtil {
     node.accept(new FunctionFinder(language, recursive))
   }
 
+  /**
+    * Checks whether the specified function is python function.
+    *
+    * @param function the function to check.
+    * @return true if it is python function.
+    */
+  def isPythonFunction(function: UserDefinedFunction): Boolean =
```

Comment:
Change this to
```
  def isPythonCall(rexCall: RexCall): Boolean = rexCall.getOperator match {
    case sfc: ScalarSqlFunction => sfc.getScalarFunction.isInstanceOf[PythonFunction]
    case _ => false
  }
```
so that we can simply the logic in `FunctionFinder` with:
```
  private class FunctionFinder(findPythonFunction: Boolean, recursive: Boolean)
    extends RexDefaultVisitor[Boolean] {

    override def visitCall(call: RexCall): Boolean = {
      findPythonFunction == isPythonCall(call) ||
        (recursive && call.getOperands.exists(_.accept(this)))
    }

    override def visitNode(rexNode: RexNode): Boolean = false
  }
```
And also can simplify the visitCall method in `ScalarFunctionSplitter` with:
```
  override def visitCall(call: RexCall): RexNode = {
    visit(isPythonCall(call), call)
  }
```